crates/core/src/test.rs

```diff
@@ -15305,3 +15305,789 @@ fn or_file() {
+                    |"#
+            .trim_margin()
+            .unwrap(),
+            expected: r#"
+                    |var message = $"{name}";
+                    |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_null_check() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                    |language csharp
+                    |
+                    |`if ($var == null) $body` => `if ($var is null)
+                    |    $body`
+                    |"#
+            .trim_margin()
+            .unwrap(),
+            source: r#"
+                    |if (customer == null)
+                    |    throw new ArgumentNullException();
+                    |"#
+            .trim_margin()
+            .unwrap(),
+            expected: r#"
+                    |if (customer is null)
+                    |    throw new ArgumentNullException();
+                    |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_switch_expression() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`switch ($value) {
+                |    case $type1 $var1:
+                |        return $result1;
+                |    case $type2 $var2:
+                |        return $result2;
+                |    default:
+                |        return $default;
+                |}` => `return $value switch {
+                |    $type1 $var1 => $result1,
+                |    $type2 $var2 => $result2,
+                |    _ => $default
+                |};`
+                |"#
+            .trim_margin()
+            .unwrap(),
+            source: r#"
+                |switch (shape) {
+                |    case Circle c:
+                |        return Math.PI * c.Radius * c.Radius;
+                |    case Rectangle r:
+                |        return r.Width * r.Height;
+                |    default:
+                |        return 0;
+                |}
+                |"#
+            .trim_margin()
+            .unwrap(),
+            expected: r#"
+                |return shape switch {
+                |    Circle c => Math.PI * c.Radius * c.Radius,
+                |    Rectangle r => r.Width * r.Height,
+                |    _ => 0
+                |};
+                |"#
+            .trim_margin()
+            .unwrap(),
+        }
+    })
+    .unwrap();
+}
+
+#[test]
+fn csharp_record_declaration() {
+    run_test_expected({
+        TestArgExpected {
+            pattern: r#"
+                |language csharp
+                |
+                |`public class $name {
+                |    public $name($params) {
+                |        $assignments
+                |    }
+                |
+                |    $_
+                |    $_
```

Comment:
For future tests we usually want to be a bit less strict on shape. I'd personally want to see `$...` work here.

Reply:
fixed in cee3e5f98f16f83de3242e944c966dbccfe7f26a